src/blockchain/src/network.rs

```diff
@@ -0,0 +1,70 @@
+/*
+   Copyright 2021 JFrog Ltd
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+use libp2p::{
+    floodsub::{Floodsub, FloodsubEvent},
+    mdns::{Mdns, MdnsEvent},
+    swarm::NetworkBehaviourEventProcess,
+    NetworkBehaviour,
+};
+
+use super::*;
+
+#[derive(NetworkBehaviour)]
+#[behaviour(event_process = true)]
+
+pub struct MyBehaviour {
```

Comment:
I think it's a good time to rename this!

Maybe just something simple?

```suggestion
pub struct Behaviour {
```